src/MSRIOGroup.cpp

```diff
@@ -285,6 +288,82 @@ namespace geopm
         }
     }
 
+    void MSRIOGroup::register_rdt_signals(void)
+    {
+        // It may be necessary to wrap the initial rdt() call in this check
+        int domain = signal_domain_type("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL");
+        int num_domain = m_platform_topo.num_domain(domain);
+        double disable = 0.0;
+        for (int dom_idx = 0; dom_idx < num_domain; ++dom_idx) {
+            disable += read_signal("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL", domain, dom_idx);
+        }
+        if ((disable != 0.0) || (!m_rdt.rdt_support)) {
+            return;
+        }
+
+        std::string signal_name = "QM_CTR_SCALED";
+        std::string msr_name = "MSR::QM_CTR:RM_DATA";
+        std::string description = "Resource Monitor Data scaled to number of bytes by the conversion factor";
+
+        auto read_it = m_signal_available.find(msr_name);
+        if (read_it != m_signal_available.end()) {
+            auto readings = read_it->second.signals;
+            int ctr_domain = read_it->second.domain;
+
+            // Begin bit, end bit, and function are currently hardcoded as there's no accessor
+            // for MSRFieldSignal values that doesn't involve reparsing the JSON file
+            MSRIOGroup::add_msr_field_signal("QM_CTR",
+                                         signal_name,
+                                         ctr_domain,
+                                         0, 31,
+                                         MSR::string_to_function("scale"),
+                                         (double)m_rdt.mbm_scalar,
+                                         IOGroup::M_UNITS_NONE,
+                                         Agg::function_to_name(agg_function(msr_name)),
+                                         description,
+                                         IOGroup::M_SIGNAL_BEHAVIOR_VARIABLE);
+        }
+
+
+        // register time signal; domain board
+        std::string time_name = "MSR::TIME";
+        std::shared_ptr<Signal> time_sig = std::make_shared<TimeSignal>(m_time_zero, m_time_batch);
+        m_signal_available[time_name] = {std::vector<std::shared_ptr<Signal> >({time_sig}),
+                                         GEOPM_DOMAIN_BOARD,
+                                         IOGroup::M_UNITS_SECONDS,
+                                         Agg::select_first,
+                                         "Time in seconds used to calculate power"};
+
+        int derivative_window = 8;
```

Comment:
put these constants in the constructor maybe?

Reply:
These match the POWER derivative window (and reuse the time signal), is it reasonable to just consolidate this from register_rdt_signals and register_power_signals?

Reply:
I'm suggesting that there are two new constant members m_derivative_window and m_sleep_time, and that these be set in the constructor.  If they can be reused elsewhere, all the better